pkg/types/snapstore.go

```diff
@@ -61,6 +61,11 @@ const (
 	ExcludeSnapshotMetadataKey = "x-etcd-snapshot-exclude"
 )
 
+var (
+	// ErrSnapshotDeleteFailDueToImmutability is the error returned when the Delete call fails due to immutability
+	ErrSnapshotDeleteFailDueToImmutability = fmt.Errorf("ErrSnapshotDeleteFailDueToImmutability")
+)
```

Comment:
This is the easiest way to establish a custom error for garbage collection, for dealing with errors faced during deletion of immutable objects.
In the future this should be removed, and a more comprehensive custom error should be created.